moz_sql_parser/sql_synthesizer.py

```diff
@@ -0,0 +1,38 @@
+import re
+
+from jinja2 import Environment, PackageLoader
+
+filters = {
+    'is_list': lambda x: isinstance(x, list),
+    'is_dict': lambda x: isinstance(x, dict)
+}
+
+template_env = Environment(loader=PackageLoader('moz_sql_parser', 'templates'))
+template_env.filters.update(filters)
+template = template_env.get_template('macros.jinja2')
+
+def cleanup_string(res):
```

Comment:
Upon first glance, this was a surprising method to have: Usually, the sub-expressions have strict control over their whitespace; especially if we want to pretty print SQL in the future. 

I think this method is needed because jinja has poor whitespace control. Maybe Jinja is targeted toward producing markup, not text in general?

Reply:
This was my first attempt to use Jinja, so I'd hate for you to get the wrong impression.  I'm pretty sure it's capable of returning a nice 'clean' easy to read SQL string, it just may be beyond me.

As for the cleanup_string function, I just did that to make it easier to compare the SQL statements at a glance.  I think the better solution would be to clean up the template, but that might make it even harder to read.